core/src/codegen/field.rs

```diff
@@ -7,6 +7,18 @@ use syn::{Ident, Path, Type};
 use crate::codegen::{DefaultExpression, PostfixTransform};
 use crate::usage::{self, IdentRefSet, IdentSet, UsesTypeParams};
 
+fn is_option(ty: &Type) -> bool {
```

Comment:
This is something `derive_builder` and `serde` also do. Should it be extracted to be part of `darling::util`?